web/client/components/map/leaflet/Layer.jsx

```diff
@@ -185,7 +185,9 @@ class LeafletLayer extends React.Component {
         if (newLayer) {
             this.removeLayer();
             this.layer = newLayer;
-            this.addLayer();
+            if (newProps.options.visibility) {
+                this.addLayer();
+            }
```

Comment:
Why do we need this change?

Reply:
@allyoucanmap  because if an edit is made to a BG layer that is not visible it was added to the map, even if it was not selected. see below

![recording_1](https://user-images.githubusercontent.com/14953970/161293584-55316789-de1a-428e-9f08-ba38fff0a60f.gif)